modules/ip6/control/gr_ip6_control.h

```diff
@@ -40,4 +40,18 @@ struct hoplist *ip6_addr_get_all(uint16_t iface_id);
 // determine if the given interface is member of the provided multicast address group
 struct nexthop *ip6_mcast_get_member(uint16_t iface_id, const struct rte_ipv6_addr *mcast);
 
+static inline int ip6_addr_linklocal_scope(struct rte_ipv6_addr *ip, uint16_t iface_id) {
+	if (!rte_ipv6_addr_is_linklocal(ip))
+		return errno_set(EFAULT);
+	((uint16_t *)ip->a)[1] = iface_id;
```

Comment:
It would be best to change this to network order to ensure consistent behavior across architectures.

```c
rte_be16_t scope_id = htons(iface_id);
// rte_ipv6_addr is aligned to 1, avoid alignment issues with casting to uint16_t
ip->a[2] = 0xff & (scope_id >> 8);
ip->a[3] = 0xff & scope_id;
```